homeassistant/components/iss/binary_sensor.py

```diff
@@ -23,88 +25,58 @@
+    async_add_entities([IssBinarySensor(coordinator, name, show_on_map)])
 
 
-class IssBinarySensor(BinarySensorEntity):
+class IssBinarySensor(
+    CoordinatorEntity[DataUpdateCoordinator[IssData]], BinarySensorEntity
+):
     """Implementation of the ISS binary sensor."""
 
     _attr_device_class = DEFAULT_DEVICE_CLASS
 
-    def __init__(self, iss_data, name, show):
+    def __init__(
+        self, coordinator: DataUpdateCoordinator[IssData], name: str, show: bool
+    ) -> None:
         """Initialize the sensor."""
-        self.iss_data = iss_data
+        super().__init__(coordinator)
         self._state = None
         self._attr_name = name
         self._show_on_map = show
 
     @property
     def is_on(self) -> bool:
         """Return true if the binary sensor is on."""
-        return self.iss_data.is_above if self.iss_data else False
+        return self.coordinator.data.is_above is True
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes."""
-        if self.iss_data:
-            attrs = {
-                ATTR_ISS_NUMBER_PEOPLE_SPACE: self.iss_data.number_of_people_in_space,
-                ATTR_ISS_NEXT_RISE: self.iss_data.next_rise,
-            }
-            if self._show_on_map:
-                attrs[ATTR_LONGITUDE] = self.iss_data.position.get("longitude")
-                attrs[ATTR_LATITUDE] = self.iss_data.position.get("latitude")
-            else:
-                attrs["long"] = self.iss_data.position.get("longitude")
-                attrs["lat"] = self.iss_data.position.get("latitude")
-
-            return attrs
-
-    def update(self):
-        """Get the latest data from ISS API and updates the states."""
-        self.iss_data.update()
-
-
-class IssData:
-    """Get data from the ISS API."""
-
-    def __init__(self, latitude, longitude):
-        """Initialize the data object."""
-        self.is_above = None
-        self.next_rise = None
-        self.number_of_people_in_space = None
-        self.position = None
-        self.latitude = latitude
-        self.longitude = longitude
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    def update(self):
-        """Get the latest data from the ISS API."""
-        try:
-            iss = pyiss.ISS()
-            self.is_above = iss.is_ISS_above(self.latitude, self.longitude)
-            self.next_rise = iss.next_rise(self.latitude, self.longitude)
-            self.number_of_people_in_space = iss.number_of_people_in_space()
-            self.position = iss.current_location()
-        except (HTTPError, requests.exceptions.ConnectionError):
-            _LOGGER.error("Unable to retrieve data")
-            return False
+        attrs = {
+            ATTR_ISS_NUMBER_PEOPLE_SPACE: self.coordinator.data.number_of_people_in_space,
+            ATTR_ISS_NEXT_RISE: self.coordinator.data.next_rise,
```

Comment:
Does the next rise datetime serialize correctly for the frontend?

Reply:
It worked before and should continue to do so. I haven't changed anything other then wrap the former `IssData` class into a coordinator. But this does not really matter, as i will remove this in the PR as this part of the API is gone anyway 😃

Reply:
Ok.